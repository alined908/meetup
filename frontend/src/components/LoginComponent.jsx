import React, {Component} from 'react';
import AuthenticationService from "../accounts/AuthenticationService.js";
import axios from 'axios';

class LoginComponent extends Component {

    constructor(props){
        super(props)
        this.state = {
            email: '',
            password: '',
            hasLoginFailed: false,
            showSuccessMessage: false
        }
        this.handleChange = this.handleChange.bind(this);
        this.loginClicked = this.loginClicked.bind(this);
    }

    handleChange(event){
        this.setState({
            [event.target.name]:event.target.value
        })
    }

    loginClicked(){
        axios.post('http://localhost:8000/token-auth/', {
            email: this.state.email,
            password: this.state.password
        })
        .then(response => {
            if (response.status === 200) {
                AuthenticationService.registerSuccessfulLogin(response.token);
                this.props.history.push("/welcome")
            }
        })
        .catch(error => {
            this.setState({showSuccessMessage: false})
            this.setState({hasLoginFailed: true})
        })
    }

    render () {
        return (
            <div>
                <h1>Login</h1>
                <div className="container">
                    {this.state.hasLoginFailed && <div className="alert alert-warning">Invalid Credentials</div>}
                    {this.state.showSuccessMessage && <div>Logged in Successfully</div>}
                    Email: <input type="email" name="email" value={this.state.email} onChange={this.handleChange}></input>
                    Password: <input type="password" name="password" value={this.state.password} onChange={this.handleChange}></input>
                    <button className="btn btn-success" onClick={this.loginClicked}>Login</button>
                </div>
            </div>
        )
    }
}

export default LoginComponent