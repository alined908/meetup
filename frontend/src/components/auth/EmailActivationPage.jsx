import React, { Component } from "react";
import { axiosClient } from "../../accounts/axiosClient";
import { connect } from "react-redux";
import { addGlobalMessage } from "../../actions/";
import { CircularProgress } from "@material-ui/core";

class EmailActivationPage extends Component {
  async componentDidMount() {
    try {
      const data = {
        uid: this.props.match.params.uid,
        token: this.props.match.params.token,
      };
      const response = await axiosClient.post("/auth/users/activation/", data);
      this.props.addGlobalMessage("success", "Email has been confirmed.");
      console.log(response.data);
    } catch (e) {
      this.props.addGlobalMessage("error", "Error in confirmation.");
      console.log(e.response);
    }
    this.props.history.push("/login");
  }

  render() {
    return (
      <div class="loading">
        <CircularProgress size={50} />
      </div>
    );
  }
}

const mapDispatchToProps = {
  addGlobalMessage,
};

export default connect(null, mapDispatchToProps)(EmailActivationPage);
