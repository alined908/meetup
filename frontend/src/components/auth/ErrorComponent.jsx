import React, {Component} from 'react';
import {ReactComponent as AccidentIcon} from "../../assets/svgs/accident.svg"
import {history} from '../MeetupApp'

class ErrorComponent extends Component {
    handleClick = () => {
        history.goBack()
    }

    render () {
        return(
            <div className="error-page">
                <div className="error-page-icon"><AccidentIcon width="100%" height="100%"/></div>
                <div className="error-page-message">
                    <div>&#x2639; - Error code: 404</div>
                    <div>We weren't able to find the page!</div>
                    <div onClick={this.handleClick}>Try Again</div>
                </div>
            </div>
        )
    }
}

export default ErrorComponent