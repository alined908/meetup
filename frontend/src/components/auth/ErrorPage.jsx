import React, {Component} from 'react';
import {ReactComponent as AccidentIcon} from "../../assets/svgs/accident.svg"
import {history} from '../MeetupApp'
import styles from '../../styles/general.module.css'

class ErrorPage extends Component {
    handleClick = () => {
        history.goBack()
    }

    render () {
        return(
            <div className={styles.errorPage}>
                <div className={styles.errorPageIcon}>
                    <AccidentIcon width="100%" height="100%"/>
                </div>
                <div className={styles.errorPageMessage}>
                    <div>Error code: 404</div>
                    <div>We weren't able to find the page!</div>
                    <div onClick={this.handleClick}>Try Again</div>
                </div>
            </div>
        )
    }
}

export default ErrorPage