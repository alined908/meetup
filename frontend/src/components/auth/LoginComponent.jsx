import React, {Component} from 'react';
import {reduxForm, Field} from 'redux-form';
import {compose} from 'redux';
import {connect} from 'react-redux';
import {signin} from "../../actions/index"
import {Typography, Paper, Grid, Fab} from '@material-ui/core';
import renderTextField from '../renderTextField'

class LoginComponent extends Component {

    onSubmit = formProps => {
        this.props.signin(formProps, () => {
            this.props.history.push("/welcome")
        });
    }

    render(){
        const {handleSubmit} = this.props;

        return (
            <div className="inner-wrap">
                <div className="inner-header">
                <Typography variant="h5">Login</Typography>
                </div>
                <div className="form">
                    <Paper elevation={3} style={{padding: "2rem 3rem"}}>
                        <form onSubmit={handleSubmit(this.onSubmit)}>
                            <Grid container style={{padding: "1rem"}} spacing={3}>
                                <Grid item xs={12}>
                                    <Field name="email" component={renderTextField} label="Email"/>
                                </Grid>
                                <Grid item xs={12}>
                                    <Field name="password" type="password" component={renderTextField} label="Password"/>
                                </Grid> 
                                <Grid item xs={12}>
                                    <div>{this.props.errorMessage}</div>
                                </Grid>
                                <Grid item xs={12}>
                                <Fab type="submit" variant="extended" color="primary" aria-label="login">Login</Fab>
                                </Grid>
                            </Grid>
                        </form>
                    </Paper>
                </div>
            </div>
        )
    }
}

function mapStatetoProps(state) {
    return {errorMessage: state.user.errorMessage}
}

const mapDispatchToProps = {
    signin
}

export default compose (
    connect(mapStatetoProps, mapDispatchToProps),
    reduxForm({form: 'signin'})
)(LoginComponent);