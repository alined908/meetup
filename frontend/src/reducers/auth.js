import {AUTH_USER, AUTH_ERROR} from "../constants/action-types"

export default function(state = {authenticated: '', errorMessage: ''}, action){
    switch(action.type){
        case AUTH_USER:
            let user = Object.values(action.payload.user)[0]
            return {...state, authenticated: action.payload.token, user: user};
        case AUTH_ERROR:
            return {...state, errorMessage: action.payload};
        default:
            return state;
    }
}   