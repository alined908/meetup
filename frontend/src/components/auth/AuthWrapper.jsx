import React, {Component} from 'react'
import {Dialog} from '@material-ui/core'
import {LoginPage} from '../components'
import {withRouter} from 'react-router-dom'

class AuthWrapper extends Component {

    constructor(props){
        super(props);
        this.state = {
            loginForm: false
        }
    }

    handleClick = event => {
        this.setState({loginForm: true})
    }

    handleClose = event => {
        event.stopPropagation();
        this.setState({loginForm: false})
    }

    render = () => {

        return (
            <div className="auth-wrapper" onClick={this.props.authenticated ? undefined : this.handleClick}>
                <div className={`${!this.props.authenticated && 'block'}`}>
                    {this.props.children}
                </div>
                <Dialog className='auth-alert' open={this.state.loginForm} fullWidth={true} onClose={this.handleClose}>
                    <LoginPage {...this.props}/>
                </Dialog>
            </div>
        )
    }
}

export default withRouter(AuthWrapper)