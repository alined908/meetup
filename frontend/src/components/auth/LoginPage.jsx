import React, { Component } from "react";
import { reduxForm, Field } from "redux-form";
import { compose } from "redux";
import { connect } from "react-redux";
import { signin, signinHelper} from "../../actions";
import { Paper, Grid, Fab } from "@material-ui/core";
import FacebookIcon from '@material-ui/icons/Facebook';
import { Link } from "react-router-dom";
import { renderTextField } from "../components";
import {axiosClient} from '../../accounts/axiosClient'
import { ReactComponent as Fan } from "../../assets/svgs/fans.svg";
import styles from "../../styles/form.module.css";
import { GoogleLogin } from 'react-google-login';
import FacebookLogin from 'react-facebook-login'
import { Helmet } from "react-helmet";
import PropTypes from "prop-types";

const validate = (values) => {
  const errors = {};
  if (!values.email) {
    errors.email = "Email is required.";
  }

  if (!values.password) {
    errors.password = "Password is required.";
  }

  return errors;
};

class LoginPage extends Component {
  handleFacebookSocialAuth = async (res) => {
    if (res.error) {
      return;
    }
    console.log(res)
    const response = await axiosClient.post(
      '/auth/facebook/', {accessToken: res.accessToken, email: res.email, name: res.name}
    )
    console.log(response.data)
    signinHelper(response.data, this.props.dispatch, () => this.props.history.push("/meetups"))
  }


  handleGoogleSocialAuth = async (res) => {
    if (res.error){
      return;
    }
    const response = await axiosClient.post(
      '/auth/google/', {tokenId: res.tokenId}
    )
    console.log(response.data)
    signinHelper(response.data, this.props.dispatch, () => this.props.history.push("/meetups"))
  }

  onSubmit = (formProps) => {
    let redirect;
    if (this.props.location.state && this.props.location.state.from) {
      redirect = () => {
        this.props.history.push(this.props.location.state.from);
      };
    } else {
      redirect = () => {
        this.props.history.push("/meetups");
      };
    }
    this.props.signin(formProps, redirect);
  };

  render() {
    const { handleSubmit, submitting, invalid } = this.props;

    return (
      <Paper className={styles.container} elevation={8}>
        <Helmet>
          <meta charSet="utf-8" />
          <title>Login</title>
        </Helmet>
        <div className={styles.left}>
          <Fan className="svg-shadow" height="70%" width="70%" />
        </div>
        <div className={styles.right}>
          <div className={styles.formhead}>
            <span className={styles.header}>Login</span>
          </div>
          <form className={styles.form} onSubmit={handleSubmit(this.onSubmit)}>
            <Grid container spacing={2}>
              <Grid item xs={12}>
                <Field name="email" component={renderTextField} label="Email" />
              </Grid>
              <Grid item xs={12}>
                <Field
                  name="password"
                  type="password"
                  component={renderTextField}
                  label="Password"
                />
              </Grid>
              {this.props.errorMessage && (
                <Grid item xs={12}>
                  <div className={styles.error}>{this.props.errorMessage}</div>
                </Grid>
              )}
            </Grid>
            <div className={styles.fab}>
              <Fab
                color="primary"
                type="submit"
                variant="extended"
                aria-label="login"
                disabled={submitting || invalid}
              >
                Login
              </Fab>
            </div>
          </form>
          <div className={styles.bottom}>
            <div className={styles.socials}>
              <GoogleLogin
                clientId={process.env.REACT_APP_GOOGLE_OAUTH2_KEY}
                buttonText="Continue with Google"
                onSuccess={this.handleGoogleSocialAuth}
                onFailure={this.handleGoogleSocialAuth}
                cookiePolicy={'single_host_origin'}
                className={`${styles.social} ${styles.google}`}
              />
              <FacebookLogin
                autoLoad={false}
                reauthenticate={true}
                appId={process.env.REACT_APP_FACEBOOK_OAUTH2_KEY}
                callback={this.handleFacebookSocialAuth}
                fields="name,email,picture"
                icon={<FacebookIcon/>}
                textButton="Continue with Facebook"
                cssClass={`${styles.social} ${styles.facebook}`}
              />
            </div>
            <div className={styles.action}>
              Don't Have An Account?
              <span className={styles.link}>
                <Link to="/register">Sign Up</Link>
              </span>
            </div>
            <div className={styles.action}>
              Forgot Password?
              <span className={styles.link}>
                <Link to="/password/reset">Reset</Link>
              </span>
            </div>
          </div>
        </div>
      </Paper>
    );
  }
}

LoginPage.propTypes = {
  errorMessage: PropTypes.string,
  signin: PropTypes.func.isRequired,
};

function mapStatetoProps(state) {
  return {
    errorMessage: state.user.loginErrorMessage,
  };
}

const mapDispatchToProps = {
  signin,
};

export default compose(
  connect(mapStatetoProps, mapDispatchToProps),
  reduxForm({
    form: "signin",
    validate
  })
)(LoginPage);

export { LoginPage as UnderlyingLoginPage };
