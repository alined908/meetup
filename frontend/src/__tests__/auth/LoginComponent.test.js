import React from 'react'
import {shallow} from 'enzyme';
import {UnderlyingLoginComponent} from "../../components/auth/LoginComponent"

describe("LoginComponent", () => {
    it ('renders without crashing given props', () => {
        const props = {
            errorMessage: "",
            signin: jest.fn(),
            handleSubmit: jest.fn()
        }
        const wrapper = shallow(<UnderlyingLoginComponent {...props}/>)
        expect(wrapper).toMatchSnapshot()
    })

    it('dispatches correct action on submit', () => {
        const props = {
            errorMessage: "",
            signin: jest.fn(),
            handleSubmit: jest.fn()
        }
        const mockData = {first_name: "Daniel", last_name: "Lee", email: "daniel@gmail.com", password: "password"}

        const wrapper = shallow(<UnderlyingLoginComponent {...props}/>)
        wrapper.instance().onSubmit(mockData);
        expect(props.signin.mock.calls.length).toBe(1);
    })
})