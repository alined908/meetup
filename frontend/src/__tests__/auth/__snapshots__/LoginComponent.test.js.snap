// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginComponent renders without crashing given props 1`] = `
<div
  className="inner-wrap"
>
  <div
    className="inner-header elevate"
  >
    <WithStyles(ForwardRef(Typography))
      variant="h5"
    >
      Login
    </WithStyles(ForwardRef(Typography))>
  </div>
  <div
    className="form"
  >
    <WithStyles(ForwardRef(Paper))
      className="form-paper"
      elevation={3}
    >
      <form>
        <WithStyles(ForwardRef(Grid))
          container={true}
          spacing={3}
          style={
            Object {
              "padding": "1rem",
            }
          }
        >
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <Field
              component={[Function]}
              label="Email"
              name="email"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <Field
              component={[Function]}
              label="Password"
              name="password"
              type="password"
            />
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid))
            item={true}
            xs={12}
          >
            <WithStyles(ForwardRef(Fab))
              aria-label="login"
              color="primary"
              type="submit"
              variant="extended"
            >
              Login
            </WithStyles(ForwardRef(Fab))>
          </WithStyles(ForwardRef(Grid))>
        </WithStyles(ForwardRef(Grid))>
      </form>
    </WithStyles(ForwardRef(Paper))>
  </div>
</div>
`;
